### ISSUE TITLE
Solve EASTL/string.h warning about type mismatch

### ISSUE THREAD

**Issue by egranata** [egranata/puppy#169]
Building EASTL with gcc -Wextra yields

```
EASTL/string.h:3255:55: warning: enumeral and non-enumeral type in conditional expression [-Wextra]
   return (currentCapacity <= SSOLayout::SSO_CAPACITY) ? SSOLayout::SSO_CAPACITY : (2 * currentCapacity);
```

Work with upstream EASTL to fix this issue

**Comment by egranata** [comment]
Refer to https://github.com/electronicarts/EASTL/pull/257

**Comment by egranata** [comment]
This is fixed upstream; we should try and bring a new dump of EASTL in the codebase

**Comment by egranata** [comment]
Closed by d13ae22bcb51afb3614816164287443ac872dc62
